skimage/feature/register_translation.py

```diff
@@ -191,20 +191,20 @@ def register_translation(src_image, target_image, upsample_factor=1,
     cross_correlation = np.fft.ifftn(image_product, axes=register_axes)
 
     # Locate maximum
-    flattened_CC = np.abs(cross_correlation).reshape(*broadcast_shape,
-                                                    np.product(register_shape))
-    maxima = np.unravel_index(np.argmax(flattened_CC, axis=-1), register_shape)
-    maxima = np.stack(maxima, axis=-1)
+    flat_CC = np.abs(cross_correlation).reshape(*broadcast_shape,
+                                                np.product(register_shape))
+    flat_maxima = np.argmax(flat_CC, axis=-1)
+    maxima = np.stack(np.unravel_index(flat_maxima, register_shape), axis=-1)
     midpoints = np.array([np.fix(axis_size / 2) for axis_size in register_shape])
 
     shifts = maxima - np.array(register_shape) * (maxima > midpoints)
     if upsample_factor == 1:
         if return_error:
             src_amp = np.sum(np.abs(src_freq) ** 2, axis=register_axes) \
-                             / src_freq.size
+                / np.product(register_shape)
             target_amp = np.sum(np.abs(target_freq) ** 2, axis=register_axes) \
-                                / target_freq.size
-            CCmax = cross_correlation[maxima]
+                / np.product(register_shape)
+            CCmax = flat_CC[..., flat_maxima]
```

Comment:
can you please explain the bugfix here. The changes look more stylistic than anything else.

Is there a reason why you prefer `np.product(shape)` vs `size`

Reply:
I'm trying to get just the size of the registered axes instead of the size of the entire image. I use `np.product(register_shape)` several times so it's probably worth defining `register_size = np.product(register_shape)` to make this more clear.

Reply:
Just use the numpy attribute `size` like it was in the original code:
```python
In [1]: import numpy as np                                                                                                                            

In [2]: a = np.zeros((4, 5))                                                                                                                          

In [3]: a.shape                                                                                                                                       
Out[3]: (4, 5)

In [4]: a.size                                                                                                                                        
Out[4]: 20
```

There is one commit where you say "bugfix" and it is unclear if you are refactoring, or actually fixing a bug. If you fixing a bug, It would be cool to actually port it back while you are working on this.

Reply:
My bad on my previous comment, you are right. ok. I get what is happening. yeah, maybe defining it as a seperate variable is good. The numpy guys just told me not to repeatidly use `np.product` on tuples, because the numpy coercion is quite slow.

Reply:
Sorry for the unclear commit message -- there is no bug in the original code. 

I will define it as a separate variable to avoid the repeated `np.product` calls.

Reply:
thats fine, I was just trying to better understand what you were doing. bit size review you know ;)